TumbleBitSetup/RsaPubKey.cs

```diff
@@ -31,7 +31,17 @@ public RsaPubKey(BigInteger N, BigInteger e)
         {
             _pubKey = new RsaKeyParameters(false, N, e);
         }
-        
+
+        public bool VerifyPoupardSternProof(PoupardSternProof proof, PoupardSternSetup setup)
```

Comment:
Same comment as above. 

From a dev perspective, I can understand why you put this code here.  Clearly the verification is an action that is computed over the private key.  

My question is, if we are adding stuff to RSApubkey.cs, which is an important file that is used in many projects, does this make it less likely for our code to be incorporated in bouncyCastle?   In other words: is our code more likely to be merged in bouncyCastle if we keep our additions in a separate file, rather than add it to the main RSAkey.cs file?